services/s3/src/test/java/software/amazon/awssdk/services/s3/multiauthsigv4a/Sigv4aSigningRegionSetTest.java

```diff
@@ -0,0 +1,251 @@
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.awscore.endpoints.AwsEndpointAttribute;
+import software.amazon.awssdk.awscore.endpoints.authscheme.SigV4AuthScheme;
+import software.amazon.awssdk.awscore.endpoints.authscheme.SigV4aAuthScheme;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.sync.ResponseTransformer;
+import software.amazon.awssdk.endpoints.Endpoint;
+import software.amazon.awssdk.http.HttpExecuteResponse;
+import software.amazon.awssdk.http.SdkHttpResponse;
+import software.amazon.awssdk.http.auth.aws.signer.RegionSet;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3AsyncClientBuilder;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3ClientBuilder;
+import software.amazon.awssdk.services.s3.endpoints.S3EndpointParams;
+import software.amazon.awssdk.services.s3.endpoints.S3EndpointProvider;
+import software.amazon.awssdk.testutils.EnvironmentVariableHelper;
+import software.amazon.awssdk.testutils.service.http.MockAsyncHttpClient;
+import software.amazon.awssdk.testutils.service.http.MockSyncHttpClient;
+
+@WireMockTest
+@DisplayName("S3 SigV4a Signing Region Set Tests")
+class Sigv4aSigningRegionSetTest {
+    private static final String EXAMPLE_BUCKET = "Example-Bucket";
+    private static final String EXAMPLE_RESPONSE_BODY = "Hello world";
+    private static final EnvironmentVariableHelper ENVIRONMENT_VARIABLE_HELPER = new EnvironmentVariableHelper();
+    public static final Function<InputStream, String> stringFromStream = inputStream ->
+        new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))
+            .lines()
+            .collect(Collectors.joining("\n"));
+
+    private final MockSyncHttpClient mockHttpClient = new MockSyncHttpClient();
+    private final MockAsyncHttpClient mockAsyncHttpClient = new MockAsyncHttpClient();
+
+    @AfterEach
+    void cleanUp() {
+        ENVIRONMENT_VARIABLE_HELPER.reset();
+    }
+
+    private static Stream<Arguments> regionSetConfigurations() {
```

Comment:
Do we have tests to verify which configuration takes precedence?

Reply:
Yeap ,
we have tested ClientConfig precedence in 
https://github.com/aws/aws-sdk-java-v2/blob/8e54e1662cdf1245c5c6160ea85754434a158af9/core/aws-core/src/test/java/software/amazon/awssdk/awscore/internal/auth/Sigv4aSigningRegionSetProviderTest.java#L51-L151

and Combination on ClientConfig with rest in

https://github.com/aws/aws-sdk-java-v2/blob/feature/master/multi-auth-sigv4a/test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/auth/Sigv4aSigningRegionSetTest.java